compiler/rustc_mir/src/transform/inline.rs

```diff
@@ -607,13 +607,9 @@ impl Inliner<'tcx> {
                 }
 
                 // Insert all of the (mapped) parts of the callee body into the caller.
-                caller_body.local_decls.extend(
-                    // FIXME(eddyb) make `Range<Local>` iterable so that we can use
-                    // `callee_body.local_decls.drain(callee_body.vars_and_temps())`
-                    callee_body
-                        .vars_and_temps_iter()
-                        .map(|local| callee_body.local_decls[local].clone()),
-                );
+                caller_body
+                    .local_decls
+                    .extend(callee_body.local_decls.drain(callee_body.arg_count + 1..)); // vars and temps
```

Comment:
So, I don't think "exposing" the internal logic of `arg_count + 1..` is all that great. Could you add a new function on `Body`, `drain_vars_and_temps`?

Reply:
Agreed